catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/testsuites/generateAllEspressoTestsSuite.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+#put header
+echo "/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */\n" > AllEspressoTestsSuite.java
+
+#put packagename
+echo "package org.catrobat.catroid.uiespresso.testsuites;\n" >> AllEspressoTestsSuite.java
+
+#put imports
+find .. -name '*Test.java' | sed 's/\.\.\//import org.catrobat.catroid.uiespresso./' | sed 's/\//\./g' | sed 's/.java/;/' | sort >> AllEspressoTestsSuite.java
+
+#put annotations
+echo "import org.junit.runner.RunWith;\nimport org.junit.runners.Suite;\n\n@RunWith(Suite.class)\n@Suite.SuiteClasses({" >> AllEspressoTestsSuite.java
+
+#put classes to be imported
+find .. -name '*Test.java' | sed 's/^.*\///' | sed 's/java/class/' | sed 's/$/,/' | sed '$s/,//' | sed 's/^/		/' >> AllEspressoTestsSuite.java
+
+#put class
+echo "})\npublic class AllEspressoTestsSuite {\n}" >> AllEspressoTestsSuite.java
+
+exit
```

Comment:
ill take it! thanks..

interestingly my version with the \n's works for me.

regarding the sort order, yeah, well, thats still kind of a problem because the generated file will lead to lint error due to wrong import order, because an import of a class higher in the package hierarchy has to be higher up. this is why the androidStudio generated sort order looks like this:

import org.catrobat.catroid.uiespresso.SmokeTest;
import org.catrobat.catroid.uiespresso.annotations.FlakyTestTest;
...

and the one generated from the script looks like this:
...
import org.catrobat.catroid.uiespresso.pocketmusic.PocketMusicActivityTest;
import org.catrobat.catroid.uiespresso.SmokeTest;
import org.catrobat.catroid.uiespresso.stage.StageSimpleTest;
...

i honestly dont really know if that bash script is such a good idea, but on the other hand, i have no alternatives right now... (except writing it in python :-) )

its really a shame that jUnit4 does not have anything build in to just do that. (jUnit3 had as far as i could google up), but if that helps, googling this shows that more people are banging their heads against the wall with this...